src/tree/FunctionAppTreeItem.ts

```diff
@@ -3,47 +3,51 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { AppSettingsTreeItem, AppSettingTreeItem, deleteSite, SiteClient } from 'vscode-azureappservice';
-import { IAzureNode, IAzureParentTreeItem, IAzureTreeItem } from 'vscode-azureextensionui';
-import { ILogStreamTreeItem } from '../commands/logstream/ILogStreamTreeItem';
+import { AppSettingsTreeItem, AppSettingTreeItem, deleteSite, ISiteTreeRoot, SiteClient } from 'vscode-azureappservice';
+import { AzureParentTreeItem, AzureTreeItem } from 'vscode-azureextensionui';
 import { localize } from '../localize';
 import { nodeUtils } from '../utils/nodeUtils';
 import { FunctionsTreeItem } from './FunctionsTreeItem';
 import { FunctionTreeItem } from './FunctionTreeItem';
 import { ProxiesTreeItem } from './ProxiesTreeItem';
 import { ProxyTreeItem } from './ProxyTreeItem';
 
-export class FunctionAppTreeItem implements ILogStreamTreeItem, IAzureParentTreeItem {
+export class FunctionAppTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
     public static contextValue: string = 'azFuncFunctionApp';
     public readonly contextValue: string = FunctionAppTreeItem.contextValue;
-    public readonly client: SiteClient;
     public logStreamPath: string = '';
 
+    private readonly _root: ISiteTreeRoot;
     private _state?: string;
     private readonly _functionsTreeItem: FunctionsTreeItem;
     private readonly _appSettingsTreeItem: AppSettingsTreeItem;
     private readonly _proxiesTreeItem: ProxiesTreeItem;
     private readonly _isLinuxPreview: boolean;
 
-    public constructor(client: SiteClient, isLinuxPreview: boolean) {
-        this.client = client;
+    public constructor(parent: AzureParentTreeItem, client: SiteClient, isLinuxPreview: boolean) {
+        super(parent);
+        this._root = Object.assign({}, parent.root, { client });
         this._state = client.initialState;
-        this._functionsTreeItem = new FunctionsTreeItem(client);
-        this._appSettingsTreeItem = new AppSettingsTreeItem(client);
-        this._proxiesTreeItem = new ProxiesTreeItem(client);
+        this._functionsTreeItem = new FunctionsTreeItem(this);
+        this._appSettingsTreeItem = new AppSettingsTreeItem(this);
+        this._proxiesTreeItem = new ProxiesTreeItem(this);
         this._isLinuxPreview = isLinuxPreview;
     }
 
+    public get root(): ISiteTreeRoot {
```

Comment:
Doesn't this conflict with the existing root property?

Reply:
Nope just overrides it

Reply:
an @override doc comment tag or //override comment would be helpful.